libminifi/src/minifi-c.cpp

```diff
@@ -0,0 +1,502 @@
+    .class_properties_ = properties,
+    .dynamic_properties_ = dynamic_properties,
+    .class_relationships_ = relationships,
+    .output_attributes_ = output_attributes,
+    .supports_dynamic_properties_ = static_cast<bool>(class_description->supports_dynamic_properties),
+    .supports_dynamic_relationships_ = static_cast<bool>(class_description->supports_dynamic_relationships),
+    .inputRequirement_ = minifi::core::annotation::toString(toInputRequirement(class_description->input_requirement)),
+    .isSingleThreaded_ = static_cast<bool>(class_description->is_single_threaded),
+  };
+
+  minifi::utils::CProcessorClassDescription c_class_description{
+    .name = name_segments.back(),
+    .class_properties = properties,
+    .class_relationships = relationships,
+    .supports_dynamic_properties = description.supports_dynamic_properties_,
+    .supports_dynamic_relationships = description.supports_dynamic_relationships_,
+    .input_requirement = toInputRequirement(class_description->input_requirement),
+    .is_single_threaded = description.isSingleThreaded_,
+
+    .callbacks = class_description->callbacks
+  };
+
+  fn(description, c_class_description);
+}
+
+}  // namespace org::apache::nifi::minifi::utils
+
+extern "C" {
+
+MinifiPropertyValidator MinifiGetStandardValidator(MinifiStandardPropertyValidator validator) {
+  switch (validator) {
+    case MINIFI_ALWAYS_VALID_VALIDATOR: return reinterpret_cast<MinifiPropertyValidator>(&minifi::core::StandardPropertyValidators::ALWAYS_VALID_VALIDATOR);
+    case MINIFI_NON_BLANK_VALIDATOR: return reinterpret_cast<MinifiPropertyValidator>(&minifi::core::StandardPropertyValidators::NON_BLANK_VALIDATOR);
+    case MINIFI_TIME_PERIOD_VALIDATOR: return reinterpret_cast<MinifiPropertyValidator>(&minifi::core::StandardPropertyValidators::TIME_PERIOD_VALIDATOR);
+    case MINIFI_BOOLEAN_VALIDATOR: return reinterpret_cast<MinifiPropertyValidator>(&minifi::core::StandardPropertyValidators::BOOLEAN_VALIDATOR);
+    case MINIFI_INTEGER_VALIDATOR: return reinterpret_cast<MinifiPropertyValidator>(&minifi::core::StandardPropertyValidators::INTEGER_VALIDATOR);
+    case MINIFI_UNSIGNED_INTEGER_VALIDATOR: return reinterpret_cast<MinifiPropertyValidator>(&minifi::core::StandardPropertyValidators::UNSIGNED_INTEGER_VALIDATOR);
+    case MINIFI_DATA_SIZE_VALIDATOR: return reinterpret_cast<MinifiPropertyValidator>(&minifi::core::StandardPropertyValidators::DATA_SIZE_VALIDATOR);
+    case MINIFI_PORT_VALIDATOR: return reinterpret_cast<MinifiPropertyValidator>(&minifi::core::StandardPropertyValidators::PORT_VALIDATOR);
+    default: gsl_FailFast();
+  }
+}
+
+void MinifiRegisterProcessorClass(const MinifiProcessorClassDescription* class_description) {
+  gsl_Expects(class_description);
+
+  auto module_name = toString(class_description->module_name);
+  minifi::BundleDetails bundle{
+    .artifact = module_name,
+    .group = module_name,
+    .version = "1.0.0"
```

Comment:
I think we shouldn't keep this 1.0.0 hardcoded value, could the version be part of the class description API?